# HG changeset patch
# User jwe
# Date 749319302 0
#      Wed Sep 29 16:15:02 1993 +0000
# Node ID 4e753587fac10724f4674ed902b75c72b0eee211
# Parent  f8e230ec7745f18f1ddd75fbec43eef3287e3613
[project @ 1993-09-29 16:15:02 by jwe]
(tree_subplot_list::print (int, ostrstream&)): If the data to be
plotted is a string but it doesn't name a file, try sending it to
gnuplot directly.

diff --git a/src/pt-plot.cc b/src/pt-plot.cc
--- a/src/pt-plot.cc
+++ b/src/pt-plot.cc
@@ -108,20 +108,16 @@ tree_plot_command::eval (int print)
       if (ptr != plot_list)
 	plot_buf << ",\\\n  ";
 
       int status = ptr->print (ndim, plot_buf);
       if (status < 0)
 	return retval;
     }
 
-// We need to make sure the buffer is null-terminated because it seems
-// that the libg++-2.3 ostrstream::str() function doesn\'t guarantee
-// that it will be...
-
   plot_buf << "\n" << ends;
 
 // Just testing...
 //  char *message = plot_buf.str ();
 //  cout << "[*]" << message << "[*]\n";
 
   if (parametric_plot && ndim == 2)
     {
@@ -245,22 +241,27 @@ tree_subplot_list::print (int ndim, ostr
 	  if (data.is_string_type ())
 	    {
 	      file = tilde_expand (data.string_value ());
 	      ifstream ftmp (file);
 	      if (ftmp)
 		{
 		  plot_buf << " \"" << file << '"';
 		  free (file);
-		  goto have_existing_file;
+		  goto have_existing_file_or_command;
 		}
 	      else
 		{
 		  free (file);
 		  file = (char *) NULL;
+
+// Opening as a file failed.  Let's try passing it along as a plot
+// command.
+		  plot_buf << " " << data.string_value ();
+		  goto have_existing_file_or_command;
 		}
 	    }
 
 	  nc = data.columns ();
 	  switch (ndim)
 	    {
 	    case 2:
 	      file = save_in_tmp_file (data, ndim);
@@ -280,17 +281,17 @@ tree_subplot_list::print (int ndim, ostr
 	    }
 	}
       else
 	return -1;
     }
   else
     return -1;
 
- have_existing_file:
+ have_existing_file_or_command:
 
   if (using != (tree_subplot_using *) NULL)
     {
       int status = using->print (ndim, nc, plot_buf);
       if (status < 0)
 	return -1;
     }
 
